TodoApi/Extensions/SerilogExtensions.cs

```diff
@@ -0,0 +1,132 @@
+
+namespace TodoApi;
+
+public static class SerilogExtensions
+{
+    public static WebApplicationBuilder AddSerilog(
+        this WebApplicationBuilder builder,
+        string sectionName = "Serilog",
+        Action<LoggerConfiguration>? extraConfigure = null)
+    {
+        builder.Services.AddOptions<SerilogOptions>()
+            .BindConfiguration(sectionName)
+            .ValidateDataAnnotations();
+
+
+        builder.Host.UseSerilog((context, serviceProvider, loggerConfiguration) =>
+        {
+            var loggerOptions = serviceProvider.GetRequiredService<IOptions<SerilogOptions>>().Value;
+
+            // https://github.com/serilog/serilog-settings-configuration
+            loggerConfiguration.ReadFrom.Configuration(context.Configuration, sectionName: sectionName);
+
+            extraConfigure?.Invoke(loggerConfiguration);
+
+            var levelString = context.Configuration.GetValue<string>($"{sectionName}:MinimumLevel:Default") ??
+                        nameof(LogEventLevel.Information);
+
+            Enum.TryParse<LogEventLevel>(levelString, true, out var logLevel);
+
+            loggerConfiguration
+                .Enrich.WithProperty("Application", "TodoApi")
+                .Enrich.FromLogContext()
+                .Enrich.WithEnvironmentName()
+                .Enrich.WithMachineName()
+                // https://rehansaeed.com/logging-with-serilog-exceptions/
+                .Enrich.WithExceptionDetails();
+
+            if (context.HostingEnvironment.IsDevelopment())
+            {
+                // https://github.com/serilog/serilog-sinks-async
+                loggerConfiguration.WriteTo.Async(writeTo => writeTo.Console(
+                    logLevel,
+                    loggerOptions.LogTemplate
+                ));
+            }
+
+            if (!string.IsNullOrEmpty(loggerOptions.ElasticSearchUrl))
+            {
+                // https://github.com/serilog-contrib/serilog-sinks-elasticsearch
+                loggerConfiguration.WriteTo.Async(
+                    writeTo =>
+                        writeTo.Elasticsearch(ConfigureElasticSink(
+                            loggerOptions.ElasticSearchUrl,
+                            context.HostingEnvironment.EnvironmentName)));
+            }
+
+            if (!string.IsNullOrEmpty(loggerOptions.SeqUrl))
+            {
+                loggerConfiguration.WriteTo.Async(writeTo => writeTo.Seq(loggerOptions.SeqUrl));
+            }
+
+            if (!string.IsNullOrEmpty(loggerOptions.LogPath))
+            {
+                loggerConfiguration.WriteTo.File(
+                    loggerOptions.LogPath,
+                    outputTemplate: loggerOptions.LogTemplate,
+                    rollingInterval: RollingInterval.Day,
+                    rollOnFileSizeLimit: true);
+            }
+        });
+
+        return builder;
+    }
+
+    private static ElasticsearchSinkOptions ConfigureElasticSink(string elasticUrl, string environment)
```

Comment:
I would remove elastic search and stick to SEQ and console.

Reply:
Ok I remove elastic

Reply:
I think elastic is fine, you can declare it inline. outside of this function though.